frontend/public/components/monitoring/targets.tsx

```diff
@@ -52,7 +52,8 @@ const ServiceMonitor: React.FC<{ target: Target }> = ({ target }) => {
     monitors,
     ({ spec }) =>
       service &&
-      new LabelSelector(spec.selector).matchesLabels(service.metadata.labels ?? {}) &&
+      (spec.selector.matchLabels === undefined ||
```

Comment:
Don't we need to handle `matchExpressions` as well? I believe `LabelSelector` supports `matchExpressions`.

Reply:
Thanks. I created a follow up PR to handle this: https://github.com/openshift/console/pull/10877